docs(platform): retire Self-Managed Enterprise documentation (#80509)
airbytehq/airbyte: docs/platform/cloud/managing-airbyte-cloud/manage-data-residency.md

- **Cloud**: If you're a cloud customer, you can choose from Airbyte's managed regions. If you're an [Enterprise Flex](../../enterprise-flex/) customer, you can also choose one of your own self-managed regions.

> - **Self-Managed**: If you're a self-managed customer, you can choose one of your own self-managed regions.

Author: ```suggestion
- **Self-Managed**: If you're an Airbyte Core user, you can choose one of your own self-managed regions.
```
